### PR TITLE
Rest and GraphQL get file implementation

### PR DESCRIPTION
**Pull request by erezmus**
- Implements the functionality to get a file in Rest and GraphQL
- Also for getting file meta data
- Added tests
- Some refactoring

### CODE REVIEW

File: client/src/GraphQLFileRetriever.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        $fileRecord = $this->retrieveFileMeta($path);
+
+        return $this->restFileRetriever->retrieveFile($fileRecord->getLink());
+    }
+
+    public function retrieveFileMeta(string $path): FileRecord
+    {
+        $query = <<<'QUERY'
+        query GetFileMeta(key: String) {
```
Comment (giorgiosironi) [inline client/src/GraphQLFileRetriever.php:38]: is this something that has to be in sync with the server or can it be a subset of fields?

Comment (erezmus) [reply client/src/GraphQLFileRetriever.php:38]: you mean the key?

Comment (erezmus) [reply client/src/GraphQLFileRetriever.php]: or do you mean the graphql schema?

Comment (erezmus) [reply client/src/GraphQLFileRetriever.php:38]: or do you mean the schema?

**Review by davidcmoulton — APPROVED** [review]

**Pull request merged**